portal-impl/test/unit/com/liferay/portal/model/PortletInstanceTest.java

```diff
@@ -27,6 +27,139 @@
 public class PortletInstanceTest {
 
 	@Test
+	public void testFromPortletNameAndFullInstanceId() {
+		PortletInstance portletInstance =
+			PortletInstance.fromPortletNameAndFullInstanceId(
+				PortletKeys.TEST, "1234_xyz");
+
+		Assert.assertEquals(PortletKeys.TEST, portletInstance.getPortletName());
+		Assert.assertEquals("xyz", portletInstance.getInstanceId());
+		Assert.assertEquals(1234, portletInstance.getUserId());
+		Assert.assertEquals("1234_xyz", portletInstance.getFullInstanceId());
+	}
+
+	@Test
+	public void testFromPortletNameAndFullInstanceIdEmpty() {
+		PortletInstance portletInstance =
+			PortletInstance.fromPortletNameAndFullInstanceId(
+				PortletKeys.TEST, "");
+
+		Assert.assertEquals(PortletKeys.TEST, portletInstance.getPortletName());
+		Assert.assertEquals(null, portletInstance.getInstanceId());
+		Assert.assertEquals(0, portletInstance.getUserId());
+		Assert.assertEquals("", portletInstance.getFullInstanceId());
```

Comment:
getFullInstanceId() should behave the same way as getInstanceId() when it has no value. Either both return null or both return an empty String. 

Please change one of the two to make them consistent.